src/snowflake/cli/_plugins/snowpark/commands.py

```diff
@@ -363,21 +372,52 @@ def build(
 
             if any(temp_deps_dir.path.iterdir()):
                 cli_console.step(f"Creating {project_paths.dependencies.name}")
+                dep_artifact = project_paths.get_dependencies_artefact()
                 zip_dir(
                     source=temp_deps_dir.path,
-                    dest_zip=project_paths.dependencies,
+                    dest_zip=dep_artifact.post_build_path,
                 )
             else:
                 cli_console.step(f"No external dependencies.")
 
     artifacts = set()
-    for entity in get_snowpark_entities(pd).values():
-        artifacts.update(entity.artifacts)
+    if FeatureFlag.ENABLE_SNOWPARK_NEW_BUILD.is_enabled():
+        for entity in get_snowpark_entities(pd).values():
+            for artifact in entity.artifacts:
+                artifacts.add(project_paths.get_artefact_dto(artifact))
+
+        with cli_console.phase("Preparing artifacts for source code"):
+            for artefact in artifacts:
+                bundle_map = BundleMap(
+                    project_root=artefact.project_root,
+                    deploy_root=(project_paths.project_root / "output").absolute(),
+                )
+                bundle_map.add(
+                    PathMapping(src=str(artefact.path), dest=(artefact.dest or None))
+                )
 
-    with cli_console.phase("Preparing artifacts for source code"):
-        for artefact in artifacts:
-            artefact_dto = project_paths.get_artefact_dto(artefact)
-            artefact_dto.build()
+                if artefact.path.is_file():
+                    for (absolute_src, absolute_dest) in bundle_map.all_mappings(
+                        absolute=True, expand_directories=False
+                    ):
+                        symlink_or_copy(
+                            absolute_src,
+                            absolute_dest,
+                            deploy_root=bundle_map.deploy_root(),
+                        )
```

Comment:
how about `bundle_map.symlik_or_copy_all_mappings(...)`?

Reply:
I wanted to avoid do changes in BudnleMap, there is a lot of code already. This code is only for us and will be changed soon when we will move to diff and sync.